core/autoscaling.go

```diff
@@ -334,6 +335,47 @@ func (a *autoScalingGroup) replaceOnDemandInstanceWithSpot(
 	return a.detachAndTerminateOnDemandInstance(odInst.InstanceId)
 }
 
+func (a *autoScalingGroup) getInstanceTypeByTagInASG() (string, error) {
+	svc := a.region.services.autoScaling
+	input := &autoscaling.DescribeTagsInput{
+		Filters: []*autoscaling.Filter{
+			{
+				Name: aws.String("instance-type"),
+			},
+		},
+	}
+	output, err := svc.DescribeTags(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case autoscaling.ErrCodeInvalidNextToken:
+				logger.Println(autoscaling.ErrCodeInvalidNextToken, aerr.Error())
+			case autoscaling.ErrCodeResourceContentionFault:
+				logger.Println(autoscaling.ErrCodeResourceContentionFault, aerr.Error())
+			default:
+				logger.Println(aerr.Error())
```

Comment:
Is the switch case really needed? Could we always print `aerr.Code` and `aerr.Error()` in order to remove it?